src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using Thread = Internal.Runtime.Augments.RuntimeThread;
```

Comment:
I do not think that this will work on CoreCLR

Reply:
Should I add it there as `#if CORERT`?

Wasn't sure as `ExecutionContext` is in the shared partition, but the corert version has 
```
using Thread = Internal.Runtime.Augments.RuntimeThread;
```
at top

Reply:
Yes, I think there will need to be ifdefs in this file - to make it compile and to mitigate the size regression.

Reply:
https://github.com/dotnet/coreclr/pull/16129

Reply:
Also added to this PR https://github.com/dotnet/corert/pull/5324/commits/1de8b11bb217a09241ef3de7b92afa0f999435c7